### PR TITLE
Append new host to ansible inventory when creating new node

### PR DESCRIPTION
**Pull request by sukantoraymond**
Currently we delete existing ansible inventory file when creating new node

### CODE REVIEW

File: pkg/ansible/ansible.go [hunk 1]
```diff
@@ -33,6 +33,36 @@ var config []byte
...
+	for instanceID := range publicIPMap {
+		inventoryContent := fmt.Sprintf("%s_%s", constants.AWSNodeAnsiblePrefix, instanceID)
+		if cloudService == constants.GCPCloudService {
+			inventoryContent = fmt.Sprintf("%s_%s", constants.GCPNodeAnsiblePrefix, instanceID)
+		}
+		inventoryContent += " ansible_host="
+		inventoryContent += publicIPMap[instanceID]
+		inventoryContent += " ansible_user=ubuntu "
+		inventoryContent += fmt.Sprintf("ansible_ssh_private_key_file=%s", certFilePath)
+		inventoryContent += " ansible_ssh_common_args='-o StrictHostKeyChecking=no'"
```
Comment (arturrez) [inline pkg/ansible/ansible.go:54]: will be different after https://github.com/ava-labs/avalanche-cli/pull/1076

File: pkg/ansible/ansible.go [hunk 2]
```diff
@@ -33,6 +33,36 @@ var config []byte
...
+			return err
+		}
+	}
+	return nil
+}
+
+// CreateTempAnsibleHostInventory creates inventory file in temp_inventory directory to be used for
+// Ansible playbook create commands so that all create commands can be run concurrently on all newly
+// created nodes and not applied to existing nodes in the cluster
+func CreateTempAnsibleHostInventory(inventoryDirPath, certFilePath, cloudService string, publicIPMap map[string]string) error {
```
Comment (arturrez) [inline pkg/ansible/ansible.go:65]: not DRY

File: pkg/ansible/ansible.go [hunk 3]
```diff
@@ -444,24 +459,17 @@ func UpdateInventoryHostPublicIP(inventoryDirPath string, nodesWoEIP map[string]
 	if err != nil {
 		return err
 	}
-	for nodeID, ansibleHostContent := range ansibleHostMap {
+	for node, ansibleHostContent := range inventory {
+		// trim prefix aws_node / gcp_node
+		strings.Split(node, "_")
```
Comment (felipemadero) [inline pkg/ansible/ansible.go:464]: not needed

File: pkg/ansible/ansible.go [hunk 4]
```diff
@@ -444,24 +459,17 @@ func UpdateInventoryHostPublicIP(inventoryDirPath string, nodesWoEIP map[string]
 	if err != nil {
 		return err
 	}
-	for nodeID, ansibleHostContent := range ansibleHostMap {
+	for node, ansibleHostContent := range inventory {
```
Comment (felipemadero) [inline pkg/ansible/ansible.go:462]: should we name it something different that node

File: pkg/utils/common.go [hunk 5]
```diff
@@ -24,3 +25,33 @@ func SetupRealtimeCLIOutput(cmd *exec.Cmd, redirectStdout bool, redirectStderr b
 	}
 	return &stdoutBuffer, &stderrBuffer
 }
+
+// SplitKeyValueStringToMap splits a string with multiple key-value pairs separated by delimiter.
+// Delimiter must be a single character
+func SplitKeyValueStringToMap(str string, delimiter string) (map[string]string, error) {
+	kvMap := make(map[string]string)
+	if str == "" || len(delimiter) == 0 {
```
Comment (felipemadero) [inline pkg/utils/common.go:33]: check delimiter len to be != 1

**Review by felipemadero — COMMENTED** [review]
not sure we need the temp file. can't we append the new node to the existing inventory?

**Comment by felipemadero** [comment]
> > not sure we need the temp file. can't we append the new node to the existing inventory?
> 
> If we add to existing inventory, we will be rerunning the playbook commands on nodes that already have the things in the playbook installed. Extends the time for the create command

can we pass to the ansible commands the subset of nodes from inventory to execute against?

File: pkg/ansible/ansible.go [hunk 6]
```diff
@@ -33,6 +33,36 @@ var config []byte
...
+	if err != nil {
+		return err
+	}
+	defer inventoryFile.Close()
+	for instanceID := range publicIPMap {
+		inventoryContent := fmt.Sprintf("%s_%s", constants.AWSNodeAnsiblePrefix, instanceID)
+		if cloudService == constants.GCPCloudService {
+			inventoryContent = fmt.Sprintf("%s_%s", constants.GCPNodeAnsiblePrefix, instanceID)
+		}
+		inventoryContent += " ansible_host="
```
[the remark this answers is not in the document]
Comment (felipemadero) [reply pkg/ansible/ansible.go:50]: yup but we may want to encapsulate the format of the string?

File: pkg/ansible/ansible.go [hunk 7]
```diff
@@ -33,6 +33,36 @@ var config []byte
...
+		inventoryContent += fmt.Sprintf(" ansible_ssh_private_key_file=%s", certFilePath)
+		inventoryContent += fmt.Sprintf(" ansible_ssh_common_args='%s'", constants.AnsibleSSHInventoryParams)
+		if _, err = inventoryFile.WriteString(inventoryContent + "\n"); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// CreateTempAnsibleHostInventory creates inventory file in temp_inventory directory to be used for
```
[the remark this answers is not in the document]
Comment (felipemadero) [reply pkg/ansible/ansible.go:62]: can we pass the nodes to execute against to the ansible commands? keeping only one inventory

**Review by felipemadero — APPROVED** [review]
LGTM

**Pull request merged**